astropy/table/soco.py

```diff
@@ -82,7 +82,7 @@ def add(self, key, value):
         Add a key, value pair.
         '''
         if self._unique and (key in self._nodes):
-            message = f'duplicate {key:!r} in unique index'
+            message = f'duplicate {key!r} in unique index'
```

Comment:
I have never seen `!` in string formatting, and don't know what it does, but I'm sure @taldcroft knows what he's doing.

Reply:
That gives the `repr(key)`, and one could equivalently write `'{repr(key)}'`.